easybuild/tools/filetools.py

```diff
@@ -1379,14 +1387,31 @@ def apply_regex_substitutions(paths, regex_subs, backup='.orig.eb'):
                     txt_utf8 = txt.decode(encoding='utf-8', errors='replace')
                     write_file(path, txt_utf8)
 
-                for line_id, line in enumerate(fileinput.input(path, inplace=1, backup=backup_ext)):
+                replacement_msgs = []
+                for line_nr, line in enumerate(fileinput.input(path, inplace=1, backup=backup_ext), start=1):
                     for regex, subtxt in compiled_regex_subs:
                         match = regex.search(line)
                         if match:
-                            origtxt = match.group(0)
-                            _log.info("Replacing line %d in %s: '%s' -> '%s'", (line_id + 1), path, origtxt, subtxt)
-                        line = regex.sub(subtxt, line)
+                            replacement_msgs.append("Replaced in line %d: '%s' -> '%s'" %
+                                                    (line_nr, match.group(0), subtxt))
+                            line = regex.sub(subtxt, line)
                     sys.stdout.write(line)
+                if replacement_msgs:
+                    _log.info('Applied the following substitutions to %s:\n%s', path, '\n'.join(replacement_msgs))
+                else:
+                    msg = 'Nothing found to replace in %s' % path
+                    if on_missing_match == run.ERROR:
+                        raise EasyBuildError(msg)
+                    elif on_missing_match == run.WARN:
+                        _log.warning(msg)
+                    else:
+                        msg = 'Nothing found to replace in %s' % path
+                        if on_missing_match == run.ERROR:
+                            raise EasyBuildError(msg)
+                        elif on_missing_match == run.WARN:
+                            print_warning(msg)
+                        else:
+                            _log.info(msg)
```

Comment:
@Flamefire Duplicated block of code, only `_log.info(msg)` should be under `else:`?